### DISCUSSION TITLE
Accordion ?

### DISCUSSION THREAD

**Discussion by ImaCrea** [thuliteio/doks#600] [Q&A]
Hello there,

I'm thinking about putting some "Try yourself" section in my documentation with simple tasks to achieve and if in need of help, user can open a collapsed section containing a solution. 

I thought about using using the accordion component baked into bootstrap since it's the library used by Doks. To achieve that, I'm inserting this kind of code:
```
<div class="accordion-item">
    <h2 class="accordion-header" id="headingOne">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
        Accordion Item #1
      </button>
    </h2>
    <div id="collapseOne" class="accordion-collapse collapse show" aria-labelledby="headingOne" data-bs-parent="#accordionExample">
      <div class="accordion-body">
        <strong>This is the first item's accordion body.</strong> It is shown by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions. You can modify any of this with custom CSS or overriding our default variables. It's also worth noting that just about any HTML can go within the <code>.accordion-body</code>, though the transition does limit overflow.
      </div>
    </div>
  </div>
```
The result is ok and even though it is not perfect, it can be polished with some css love.

***But*** this is going to add a ton of html into my markdown pages. Not so elegant. I was wondering if anyone here had an idea on how to achieve that in a better way? Maybe in the same spirit as alert section?

Thanks for any help :)

**Comment by h-enk** [comment]
My first idea:

- use a data file to record the accordion headers + accordion bodies
- create a layout partial generating the html, looping over the data file
- create a shortcode pulling in the layout partial

**Reply by ImaCrea** [reply]
I see, thanks for your intuition. I'm really discovering Hugo at the moment so I'll have to dig into its doc to see how I can achieve that. If you could give me a bit more details like an example or something I'd appreciate a lot.

**Reply by h-enk** [reply]
See also: https://discourse.gohugo.io/t/markdown-to-formatted-faq/26799/5

BTW, my second idea is more like this: https://discourse.gohugo.io/t/markdown-to-formatted-faq/26799/7

I'm working on a (similar) FAQ module for Doks, but it's still in the works.

**Reply by h-enk** [reply]
@ImaCrea, I've added a `<details>` shortcode to Doks — see also https://github.com/h-enk/doks/pull/661
